src/detect.c

```diff
@@ -1218,7 +1218,7 @@ static bool DetectRunTxInspectRule(ThreadVars *tv,
         } else if ((inspect_flags & DE_STATE_FLAG_FULL_INSPECT) == 0 && mpm_in_progress) {
             TRACE_SID_TXS(s->id, tx, "no need to store no-match sig, "
                     "mpm will revisit it");
-        } else {
+        } else if (inspect_flags != 0) {
```

Comment:
we store more than just the flags, we also store `file_no_match`. Can we have a case where we'd need that stored still?

Reply:
Very good point from your knowledge and wisdom. Will look that through

Reply:
Side note, feels weird to see `file_no_match` as `u16` when it is only 0 or 1 up to its use in `StoreFileNoMatchCnt` as `+= file_no_match;`

Reply:
The goal of the logic, but I wouldn't be surprised if it is broken, is to stop tracking the files if all sigs that need it definitively failed to match. So it should increment this for each unique sig that fails to match.

Reply:
> The goal of the logic, but I wouldn't be surprised if it is broken, is to stop tracking the files if all sigs that need it definitively failed to match. So it should increment this for each unique sig that fails to match.

That is what I understood from reading the code